addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -3033,10 +3037,31 @@ options.registry.CoverProperties = options.Class.extend({
      * @see this.selectClass for parameters
      */
     background: async function (previewMode, widgetValue, params) {
+        if (previewMode === false) {
+            this.$image[0].classList.remove("o_b64_image_to_save");
+        }
         if (widgetValue === '') {
             this.$image.css('background-image', '');
             this.$target.removeClass('o_record_has_cover');
         } else {
+            if (previewMode === false) {
+                const imgEl = document.createElement("img");
+                imgEl.src = widgetValue;
+                await loadImageInfo(imgEl, this._rpc.bind(this));
+                if (imgEl.dataset.mimetype && ![
+                    "image/gif",
+                    "image/svg+xml",
+                    "image/webp",
+                ].includes(imgEl.dataset.mimetype)) {
+                    // Convert to webp but keep original width.
+                    imgEl.dataset.mimetype = "image/webp";
+                    const base64src = await applyModifications(imgEl, {
+                        mimetype: "image/webp",
+                    });
+                    widgetValue = base64src;
+                    this.$image[0].classList.add("o_b64_image_to_save");
+                }
```

Comment:
couldn't you create a function for this and use it also in https://github.com/odoo/odoo/pull/148068

Reply:
This is being done in #168301 - which will convert these two calls if they are merged before.